pkg/platform/platform.go

```diff
@@ -73,6 +74,9 @@ type Platform interface {
 	// GetFunctions will list existing functions
 	GetFunctions(getFunctionsOptions *GetFunctionsOptions) ([]Function, error)
 
+	// CleanseFunctions will filter out some functions
+	CleanseFunctions(*CleanseOptions, []Function) ([]Function, error)
```

Comment:
"Cleanse" terminology is misleading IMO
Suggestions:
- filterX
- filterXByPermissions
- AuthFilterX
....